2005-05-19  Tomasz K³oczko  <kloczek@pld.org.pl>

	* src/newgrp.c:
	fixed NULL pointer dereference - getlogin() and ttyname() can
	return NULL which is not checked (http://bugs.debian.org/162303).

--- src/newgrp.c	31 Mar 2005 05:14:54 -0000	1.27
+++ src/newgrp.c	19 May 2005 11:28:27 -0000	1.28
@@ -379,8 +379,13 @@ int main (int argc, char **argv)
 		SYSLOG ((LOG_INFO, "user `%s' switched to group `%s'",
 			 name, group));
 	if (getdef_bool ("SYSLOG_SG_ENAB")) {
-		char *loginname = xstrdup (getlogin ());
-		char *tty = xstrdup (ttyname (0));
+		char *loginname = getlogin ();
+		char *tty = ttyname (0);
+
+		if (loginname != NULL)
+			loginname = xstrdup (loginname);
+		if (tty != NULL)
+			tty = xstrdup (tty);
 
 		if (loginname == NULL)
 			loginname = "???";
